nixos/modules/services/networking/nbd.nix

```diff
@@ -0,0 +1,152 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.nbd;
+  configFormat = pkgs.formats.ini { };
+  iniFields = with types; attrsOf (oneOf [ bool int float str ]);
+  serverConfig = configFormat.generate "nbd-server-config"
+    ({
+      generic = ({
+        user = "root";
+        group = "root";
+        port = cfg.server.listenPort;
+      }
+      // cfg.server.extraOptions
+      // (if (cfg.server.listenAddress != null)
+      then { listenaddr = cfg.server.listenAddress; }
+      else { }));
+    }
+    // (mapAttrs
+      (_: { path, allowAddresses, extraOptions }: {
+        exportname = path;
+      }
+      // extraOptions
+      // (if (allowAddresses != null)
+      then {
+        authfile = pkgs.writeText "authfile" ''
+          ${concatStringsSep "\n" allowAddresses}
+        '';
+      } else { }))
+      cfg.server.exports)
+    );
+  allowedDevices =
+    concatLists
+      (mapAttrsToList
+        (_: { path, ... }: if hasPrefix "/dev/" path then [ "${path} rw" ] else [ ])
+        cfg.server.exports);
+in
+{
+  options = {
+    services.nbd = {
+      server = {
+        enable = mkEnableOption "the Network Block Device (nbd) server";
+
+        listenPort = mkOption {
+          type = types.port;
+          default = 10809;
+          description = "Port to listen on. The port is NOT automatically opened in the firewall.";
+        };
+
+        extraOptions = mkOption {
+          type = iniFields;
+          default = {
+            allowlist = false;
+          };
+          description = ''
+            Extra options for the server. See
+            <citerefentry><refentrytitle>nbd-server</refentrytitle>
+            <manvolnum>5</manvolnum></citerefentry>.
+          '';
+        };
+
+        exports = mkOption {
+          description = "Files or block devices to make available over the network.";
+          default = { };
+          type = with types; attrsOf
```

Comment:
this `with` applies to all expressions inside, so any `types.` and `with types;` can be removed.

Reply:
Done.